trade-finance-manager-api/src/v1/helpers/dealTypeCoverStartDate.helper.js

```diff
@@ -0,0 +1,26 @@
+// checks for coverStartDate and returns based on deal-type as stored differently
+const dealTypeCoverStartDate = (facilitySnapshot) => {
+  const { coverStartDate } = facilitySnapshot;
+
```

Comment:
Will `coverStartDate` always be initialised when reaching this function, do we need a null check?

Reply:
changed

Reply:
Sorry I don't think this is right, what I meant was whether do we need something like this or will the cover start date (cross-product) will all be set:

```
if (!coverStartDate || !facilitySnapshot['requestedCoverStartDate-day'] !! facilitySnapshot.requestedCoverStartDate) {
    return null;
}
```

Reply:
the change I need is necessary as BSS does not have coverStartDate so it could crash.  It will be one or the other.  It will not come to this function unless there is a coverEndDate amendment.  And that amendment cannot be set unless there is an issued facility which has a coverStartDate